File: finance-api/src/main/java/com/financesystem/finance_api/modules/governance/notifications/application/dto/UpsertNotificationPreferenceRequest.java
package com.financesystem.finance_api.modules.governance.notifications.application.dto;

import com.financesystem.finance_api.modules.governance.notifications.domain.model.NotificationCategory;

public record UpsertNotificationPreferenceRequest(
        NotificationCategory category,
        boolean pushEnabled,
        boolean inAppEnabled,
        boolean emailEnabled,
        boolean smsEnabled
) {
}

File: finance_mobile/lib/infrastructure/models/upsert_notification_preference_request.dart
class UpsertNotificationPreferenceRequest {
  final String category;
  final bool pushEnabled;
  final bool inAppEnabled;
  final bool emailEnabled;
  final bool smsEnabled;

  UpsertNotificationPreferenceRequest({
    required this.category,
    required this.pushEnabled,
    required this.inAppEnabled,
    required this.emailEnabled,
    required this.smsEnabled,
  });

  Map<String, dynamic> toJson() {
    return {
      'category': category,
      'pushEnabled': pushEnabled,
      'inAppEnabled': inAppEnabled,
      'emailEnabled': emailEnabled,
      'smsEnabled': smsEnabled,
    };
  }
}
